### PR TITLE
Refactors to hide prisma config in logger

### PR DESCRIPTION
**Pull request by dthyresson**
Fixes https://github.com/redwoodjs/redwood/issues/2074

> In release 0.27.0 we've added the ability to log prisma events, by a api/src/lib/prisma.ts file which configures the way Prisma emits events and associates the user's logger to those events in order to capture them.

> My concern is that Prisma modifies their logging api and that a bunch of user's would have to update their logging code in order to upgrade to a future release.

> My proposal is such that the capturing of Prisma logging events is obscured.

This PR:

* refactors accordingly
* gets rid of prisma.ts template (yay!) is CRWA
* adds prisma client into to prisma logger
* updates README docs

fixes issue #2051

### CODE REVIEW

File: packages/create-redwood-app/template/api/src/lib/db.ts [hunk 1]
```diff
@@ -1,9 +1,21 @@
 // See https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/constructor
```
Comment (peterp) [inline packages/create-redwood-app/template/api/src/lib/db.ts:1]: 👏 👏 👏 

So nice!

**Review by peterp — APPROVED** [review]
Love it, thanks for this @dthyresson. Once the linting problems are solved this is good to go!

**Comment by peterp** [comment]
Oh, I thought we're going to try and get this in today, if not, then all's good

**Comment by peterp** [comment]
Although, I would really love to get this in before we release it.

**Comment by thedavidprice** [comment]
@peterp agreed and making sure it happens!

**Comment by thedavidprice** [comment]
@dthyresson please merge when ready and let me know when done. Great work!

**Comment by dthyresson** [comment]
Added a fix and docs for issue #2051 so that one can override prettyPrint defaults but not lose other options.

**Pull request merged**